crates/services/p2p/src/p2p_service.rs

```diff
@@ -901,27 +901,33 @@ impl FuelP2PService {
     }
 }
 
-#[allow(clippy::cast_possible_truncation)]
 #[cfg(test)]
 mod tests {
+    #![allow(non_snake_case)]
+    #![allow(clippy::cast_possible_truncation)]
```

Comment:
Perhaps unrelated to this PR but why do we need this one?

Reply:
You're right, it was already here. In general, I am in favor of allowing panickable arithmetic and dangerous casts in tests, for the same reason I'm okay with `unwrap` in tests.

I'm not sure specifically what this is enabling in the tests, but I'd encourage this pattern.

Reply:
Yeah agreed.